extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/FragmentMethodsAdder.java

```diff
@@ -34,12 +34,26 @@ public void add(ClassCreator classCreator, String generatedClassName,
             List<DotName> customInterfaceNamesToImplement, Map<String, FieldDescriptor> customImplNameToHandle) {
         for (DotName customInterfaceToImplement : customInterfaceNamesToImplement) {
             Collection<ClassInfo> knownImplementors = index.getAllKnownImplementors(customInterfaceToImplement);
-            if (knownImplementors.size() != 1) {
+            ClassInfo customInterfaceImpl = null;
+            if (knownImplementors.size() > 1) {
+                for (ClassInfo knownImplementor : knownImplementors) {
+                    if (knownImplementor.name().toString().endsWith("Impl")) { // the default suffix that Spring Data JPA looks for is 'Impl'
```

Comment:
Just to be sure, is the Spring constraint that the class must end with `Impl` or that the name is `MyInterfaceImpl`?

Reply:
Spring expects a `Impl` postfix. The name does not have to be an exact match otherwise.